src/app/components/todolist/todolist.component.ts

```diff
@@ -42,9 +43,23 @@ export class TodolistComponent implements OnInit {
 
     // clear textbox after enter
     this.todoTitle = '';
-    this.idForTodo++;
+    // find max ID for todo and set the maxid variable to be the highest id
+    var maxid = 0;
+    this.todos.forEach(function(todo){
```

Comment:
You could just write:
```
const maxId = Math.max(...this.todos)
```

Reply:
I've changed the code slightly to now use Math.max to calculate the max id.  I had to map the todo array to numbers as i was getting an error saying stating that Todos[] was not a number.  I'm not sure if there is a better solution that using map here?

Reply:
No that looks good!